korma/src/commonMain/kotlin/com/soywiz/korma/geom/shape/Shape2d.kt

```diff
@@ -139,6 +139,43 @@ inline fun VectorPath.emitPoints(flush: () -> Unit, emit: (x: Double, y: Double)
     flush()
 }
 
+@Suppress("DuplicatedCode")
+fun IPointArrayList.toRectangleOrNull(): Shape2d.Rectangle? {
+    if (this.size != 4) return null
+    val x0 = getX(0)
+    val y0 = getY(0)
+    val x1 = getX(1)
+    val y1 = getY(1)
+    val x2 = getX(2)
+    val y2 = getY(2)
+    val x3 = getX(3)
+    val y3 = getY(3)
+    if (x0 == x1 && x2 == x3 && x0 != x2) {
+        return if ((y0 == y2 && y1 == y3 && y0 != y1) || (y0 == y3 && y1 == y2 && y0 != y1))
```

Comment:
I wonder if this could be simplified, since that many ifs make it harder to follow. Here are a few proposals for this: https://stackoverflow.com/questions/2303278/find-if-4-points-on-a-plane-form-a-rectangle
So maybe checking if it is a rectangle, you can then just use min and max functions to get the bounds and construct the rectangle inline here without additional functions

Reply:
Yeah, I saw this question on stackoverflow before implementing the method. But all those answers are based on the assumption that all four points are not equal. Also they don't check where each point should be. But I agree there are too many ifs here because I tried to implement it computationally efficient. I'll see how I can simplify this solution.